lib/licensed/sources/swift.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+require "json"
+require "pathname"
+require "uri"
+
+require "licensed/sources/helpers/content_versioning"
+
+module Licensed
+  module Sources
+    class Swift < Source
+      def enabled?
+        return unless Licensed::Shell.tool_available?("swift") && swift_package?
+        File.exist?(package_resolved_file_path)
+      end
+
+      def enumerate_dependencies
+        pins.map { |pin|
+          errors = []
+
+          begin
+            name = pin["package"]
+            version = pin.dig("state", "version")
+            path = dependency_path_for_url(pin["repositoryURL"])
```

Comment:
if you don't expect to have an error from name or version can you move them outside the begin/rescue?  also, can you declare path outside of the block with a default nil value or similar?  it's a little weird to see variables declared inside of a block, and then used external to that block.